nixos/modules/config/shells-environment.nix

```diff
@@ -106,15 +106,22 @@ in
       type = types.lines;
     };
 
-    environment.shellAliases = mkOption {
-      default = {};
+    environment.shellAliases = mkOption rec {
+      default = {
+        ls = "ls --color=tty";
+        ll = "ls -l";
+        l  = "ls -alh";
+      };
       example = { ll = "ls -l"; };
       description = ''
         An attribute set that maps aliases (the top level attribute names in
         this option) to command strings or directly to build outputs. The
         aliases are added to all users' shells.
       '';
-      type = types.attrs; # types.attrsOf types.stringOrPath;
+      type = types.attrs // {
+        # merge aliases defined by the user and the defaults
+        merge = loc: defs: default // types.attrs.merge loc defs;
+      };
```

Comment:
yes, I was not sure what to do here.
But to me it seems a bit strange that if you just want to add a new alias, that the defaults disappear.

Maybe we should move these entirely to example? (this would change existing systems)
Or maybe add something like a `environment.useDefaultShellAliases` option, which is set to true by default.